CMakeSettings.json

```diff
@@ -0,0 +1,135 @@
+{
+  // See https://go.microsoft.com//fwlink//?linkid=834763 for more information about this file.
```

Comment:
Does this file really need to exist at the root of the repository?

Reply:
Yes, it's a requirement for the Visual Studio IDE experience.